domain/kaspadstate/model/readonlyutxoset.go

```diff
@@ -0,0 +1,5 @@
+package model
+
+// ReadOnlyUTXOSet ...
+type ReadOnlyUTXOSet interface {
```

Comment:
We should decide what's the API of ReadOnlyUTXOSet, SelectedParentIterator, etc.

Someone is going to implement those. Someone else is going to use those.
Therefore APIs should be ready ahead of time.